src/query/service/src/sessions/query_ctx.rs

```diff
@@ -191,26 +192,38 @@ impl QueryContext {
             .shared
             .catalog_manager
             .build_catalog(table_info.catalog_info.clone(), self.session_state())?;
-        match table_args {
-            None => {
+
+        let is_default = catalog.info().catalog_type() == CatalogType::Default;
```

Comment:
> maybe add a new `is_default ` function in catalog trait.

I don't like this idea because the catalog trait already has too many functions. Perhaps we could add a new function for CatalogInfo in the future if this pattern widely used.